pkg/controller/migplan/registry.go

```diff
@@ -99,6 +101,47 @@ func (r ReconcileMigPlan) ensureMigRegistries(plan *migapi.MigPlan) error {
 	return err
 }
 
+func (r ReconcileMigPlan) ensureRegistryHealth(plan *migapi.MigPlan) error {
+	nEnsured := 0
+	clusters, err := r.planClusters(plan)
+	if err != nil{
+		return liberr.Wrap(err)
+	}
+
+	for _, cluster := range clusters{
+
+		if !cluster.Status.IsReady() {
+			continue
+		}
+
+		client, err := cluster.GetClient(r)
+		if err != nil {
+			return liberr.Wrap(err)
+		}
+
+		registryStatusUnhealthy, err := isRegistryPodUnhealthy(plan, client)
+		if err != nil {
+			return liberr.Wrap(err)
+		}
+
+		if !registryStatusUnhealthy{
+			nEnsured++
+		}
+	}
+
+	if nEnsured == 2 {
+		plan.Status.SetCondition(migapi.Condition{
+			Type:     RegistriesHealthy,
+			Status:   True,
+			Category: migapi.Required,
+			Message:  RegistriesHealthyMessage,
```

Comment:
I would like to know if the source registry is unhealthy or destination, which pod(name and namespace) is unhealthy. Its not a blocker for now. So if you want, you can either add a todo or address this now. We would definitely need to address it before the end of next sprint

Reply:
Done, added the context details